007_team_5_agent_design/007_team_5_agent_design.tex

```diff
@@ -1 +1,522 @@
+    \item \ToDo{Check this one} It would require the agent to take less food than is required to avoid entering the ``critical'' HP band.
+\end{itemize}
+If the treaty does not meet any of the above conditions, the \texttt{decision} value is then computed as:
+\[\texttt{decision} = 6 - \texttt{selfishness} + \texttt{proposerFavour} - \frac{\texttt{treatyDuration}}{3}\]
+
+This calculation is based on the following considerations:
+\begin{itemize}
+    \item Selfish agents will always be less likely to sign treaties
+    \item The proposing agent is more likely to be trusted if the receiving agent views them more favourably
+    \item Longer treaties are viewed more negatively because factors such as the receiving agent's opinion of the proposer and the agent's selfishness can change over time, meaning the same treaty could be viewed less favourably in future. Therefore, the agent should avoid constraining its actions to satisfy an agent of which it could form a negative opinion.
+\end{itemize}
+
+If the \texttt{decision} value is positive, the treaty is accepted.
+
+\subsection*{Proposing Treaties}\label{sec:team5-treaties-proposal}
+At the start of each day, the agent checks whether it has become a `leader': if it has, then the agent proposes a treaty. Whether an agent becomes a leader or not is based on:
+\begin{itemize}
+    \item The agent's selfishness: a selfish agent is less likely to propose treaties
+    \item The agent's floor: agents higher up in the tower are more likely to propose treaties. This is because agents higher up in the tower have more access to food and therefore have more control over the food available to those below them. This control means these agents are in a better position to effect change.
+    \item The agent's \texttt{leadership} threshold: this is randomly set during the agent's initialisation and those with a lower leadership threshold are more likely to become leaders.
+\end{itemize}
+The leadership value is calculated by:
+\[ \texttt{leadership} = x - \texttt{selfishness} - \texttt{agentFloor}\]
+The variable $x$ is a random number between 3 and 12, inclusive. If \texttt{leadership} is greater than the agent's leadership threshold, then the agent becomes a leader and proposes a treaty.
+
+The proposed treaty is always that agents whose HP is greater than or equal to a certain value should always leave 100\% of the food on the platform, and always lasts for five days. The threshold value for HP is calculated using Equation \ref{eq:team5-treaty-hp-threshold}. Note that in this equation, \texttt{WeakLevel} refers to the threshold between the ``critical'' and ``weak'' HP states.
+
+\begin{equation}\label{eq:team5-treaty-hp-threshold}
+    \texttt{hpThreshold} = \texttt{aimHP} - \frac{\texttt{aimHP} - \texttt{WeakLevel}}{10}\times\left(\texttt{leadership} - \texttt{leadershipThreshold}\right)
+\end{equation}
+
```

Comment:
```suggestion
It was decided that of the treaty types available to agents in the tower, HP limiting treaties were the most suitable for composing resource management based agreements between agents. This is because treaties based directly on food amounts agreed to leave a certain amount of food. This made it difficult to construct treaties that controlled the amount of food agent took in a fair manner. While food is the main resource in the tower, it was agreed that HP could be viewed as a secondary resource, being directly affected by the main resource and reflecting the allocation of the main resource. 
```